lib/puppet/http/service.rb

```diff
@@ -47,6 +47,18 @@ def add_puppet_headers(headers)
     modified_headers
   end
 
+  def add_extra_headers(headers)
```

Comment:
@joshcooper I wasn't sure if I should just add this logic in to the `add_puppet_headers` method, or create a new method for adding the extra user-defined headers. Thoughts?

Reply:
I'm 👍 to adding to `add_puppet_headers` as they should be safe to include in all puppet REST APIs, but not something we would want to include when connecting to the forge or a customer's on-prem artificatory.